### PR TITLE
Merge into the stef_negate_matcher branch (solves issue #133)

### PR DESCRIPTION
**Pull request by StefH**

### CODE REVIEW

File: src/WireMock.Net/Matchers/Request/RequestMessageBodyMatcher.cs [hunk 1]
```diff
@@ -105,27 +105,38 @@ public double GetMatchingScore(RequestMessage requestMessage, RequestMatchResult
...
+                if (requestMessage.Body != null)
+                {
+                    return stringMatcher.IsMatch(requestMessage.Body);
+                }
+
+                // If the  body is a JSON object, try to use BodyAsStringOriginal to match.
+                if (requestMessage.BodyAsJson != null && requestMessage.BodyAsStringOriginal != null)
+                {
+                    return stringMatcher.IsMatch(requestMessage.BodyAsStringOriginal);
+                }
```
Comment (alastairtree) [inline src/WireMock.Net/Matchers/Request/RequestMessageBodyMatcher.cs]: So if we try and use Body here and remove BodyAsStringOriginal we can get by without introducing another version of the body on the requestMessage.

File: src/WireMock.Net/Util/BodyParser.cs [hunk 2]
```diff
@@ -58,6 +59,7 @@ internal static class BodyParser
             else if (contentTypeHeaderValue != null && contentTypeHeaderValue.StartsWith("application/json", StringComparison.OrdinalIgnoreCase))
             {
                 var stringData = await ReadStringAsync(stream);
+                data.BodyAsStringOriginal = stringData.Item1;
```
Comment (alastairtree) [inline src/WireMock.Net/Util/BodyParser.cs]: And then put Item1 into Body instead and remove BodyAsStringOriginal?

File: src/WireMock.Net/Matchers/Request/RequestMessageBodyMatcher.cs (hunk 1 above)
Comment (StefH) [reply src/WireMock.Net/Matchers/Request/RequestMessageBodyMatcher.cs]: Fixed

File: src/WireMock.Net/Util/BodyParser.cs (hunk 2 above)
Comment (StefH) [reply src/WireMock.Net/Util/BodyParser.cs]: Fixed

**Comment by StefH** [comment]
Review comments fixed, please double check.

**Pull request merged**